newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js

```diff
@@ -31,11 +45,13 @@ const zipProjectAndCommitVersion = async ({
   cloudProjectId: string,
   options?: {| previousVersion: string |},
 |}): Promise<?string> => {
-  const archive = await zipProject(project);
+  const zippedProject = await zipProject(project);
+  const archiveIsSane = await checkZipContent(zippedProject, project);
+  if (!archiveIsSane) return;
```

Comment:
On the contrary here, should we throw an error?

Reply:
The 3 places in the app where zipProjectAndCommitVersion is used are expecting this to throw in case of error I think. See `generateOnSaveProject` and the `wasSaved` in MainFrame. We have either:

- wasSaved === true, everything went well
- wasSaved === false, the save was *canceled* but this is still a happy path and expected (I have the right to cancel the save as a user, for example if I click cancel on a "Save as" native dialog).
- exception thrown and caught in MainFrame in case an unrecoverable error happened (typically, the case here: we were not able to save and we're not sure why but we're sure it failed)